addons/website_sale/models/product_template.py

```diff
@@ -455,7 +455,11 @@ def _search_get_detail(self, website, order, options):
         with_description = options['displayDescription']
         with_category = options['displayExtraLink']
         with_price = options['displayDetail']
-        domains = [website.sale_product_domain()]
+        domains = website.sale_product_domain()
+        if not self.env.user._is_internal():
+            domains = [expression.AND([domains, [('is_published', '=', True)]])]
+        else:
+            domains = [domains]
```

Comment:
sorry, we had discussed it and I advised to use expression.AND, but here specifically it seems that base_domain is a list of domain to which we will apply expression.AND when using them:

https://github.com/odoo/odoo/blob/452cf1c4999062ecc6d2a6e54960fbf9dff9c41b/addons/website/models/mixins.py#L334
https://github.com/odoo/odoo/blob/452cf1c4999062ecc6d2a6e54960fbf9dff9c41b/addons/website/models/website.py#L1818
https://github.com/odoo/odoo/blob/452cf1c4999062ecc6d2a6e54960fbf9dff9c41b/addons/website/models/website.py#L1851


then it would make sense to do as the other domains.append do in this file, eg:
```suggestion
        domains = [website.sale_product_domain()]
        if not self.env.user._is_internal():
            domains.append([('is_published', '=', True)])
```

Reply:
Thanks !